geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/RedundancyCommandUtils.java

```diff
@@ -0,0 +1,269 @@
+  public static final String SUMMARY_SECTION = "summary-section";
+  public static final String ZERO_REDUNDANT_COPIES =
+      "Number of regions with zero redundant copies = ";
+  public static final String PARTIALLY_SATISFIED_REDUNDANCY =
+      "Number of regions with partially satisfied redundancy = ";
+  public static final String FULLY_SATISFIED_REDUNDANCY =
+      "Number of regions with fully satisfied redundancy = ";
+  public static final String ZERO_REDUNDANCY_SECTION = "zero-redundancy";
+  public static final String UNDER_REDUNDANCY_SECTION = "under-redundancy";
+  public static final String SATISFIED_REDUNDANCY_SECTION = "satisfied-redundancy";
+  public static final String PRIMARIES_INFO_SECTION = "primaries-info";
+  public static final String EXCEPTION_MEMBER_MESSAGE = "Exception occurred on member %s: %s";
+  public static final Version REDUNDANCY_COMMAND_ADDED_VERSION = Version.GEODE_1_13_0;
+  public static final String INDENT = "  ";
+
+  void populateLists(List<RebalanceOperationPerformer.MemberPRInfo> membersForEachRegion,
+      List<String> noMemberRegions, String[] includeRegions, String[] excludeRegions,
+      InternalCache cache) {
+    // Include all regions
+    if (includeRegions == null) {
+      // Exclude these regions
+      List<String> excludedRegionList =
+          excludeRegions != null ? Arrays.asList(excludeRegions) : new ArrayList<>();
+
+      List<RebalanceOperationPerformer.MemberPRInfo> memberRegionList =
+          getMembersForEachRegion(excludedRegionList, cache);
+      membersForEachRegion.addAll(memberRegionList);
+    } else {
+      for (String regionName : includeRegions) {
+        DistributedMember memberForRegion = getOneMemberForRegion(regionName, cache);
+
+        // If we did not find a member for this region name, add it to the list of regions with no
+        // members
+        if (memberForRegion == null) {
+          noMemberRegions.add(regionName);
+        } else {
+          RebalanceOperationPerformer.MemberPRInfo memberPRInfo =
+              new RebalanceOperationPerformer.MemberPRInfo();
+          memberPRInfo.region = regionName;
+          memberPRInfo.dsMemberList.add(memberForRegion);
+          membersForEachRegion.add(memberPRInfo);
+        }
+      }
+    }
+  }
+
+  // Extracted for testing
+  List<RebalanceOperationPerformer.MemberPRInfo> getMembersForEachRegion(
+      List<String> excludedRegionList, InternalCache cache) {
+    return RebalanceOperationPerformer.getMemberRegionList(
+        ManagementService.getManagementService(cache), cache, excludedRegionList);
+  }
+
+  // Extracted for testing
+  DistributedMember getOneMemberForRegion(String regionName, InternalCache cache) {
+    String regionNameWithSeparator = regionName;
+    // The getAssociatedMembers method requires region names start with '/'
+    if (!regionName.startsWith("/")) {
+      regionNameWithSeparator = "/" + regionName;
```

Comment:
Maybe use `CacheConfig.SEPARATOR` instead of hardcoding the character? (I know we don't it consistently across the code, but...).

Reply:
Good call